get stake snapshots using pallas observer

falcucci opened the pull request:
fixes #2

falcucci committed:
add pallas stake snapshots integration

falcucci committed:
minor changes

falcucci committed:
refactor: code and improve test robustness

- remove a print statement for "computed_stake_distribution"

scarmuega commented on mithril-common/src/chain_observer/pallas_observer.rs:
```diff
@@ -175,6 +181,62 @@ impl PallasChainObserver {
+
+        let stake_snapshot = self.do_stake_snapshots_state_query(statequery).await?;
+
+        let mut stake_distribution = StakeDistribution::new();
+
+        for (key, stakes) in stake_snapshot.snapshots.stake_snapshots.iter() {
+            if stakes.snapshot_mark_pool > 0 {
+                let (pool_hash, stake) = self.get_stake_pool_hash(key, stakes)?;
+                stake_distribution.insert(pool_hash, stake);
+            }
+        }
+
+        Ok(Some(stake_distribution))
```
why are we passing the `stakes` value to the `get_stake_pool_hash` only to get the `snapshot_mark_pool` value back?

wouldn't it make more sense to just do:

```
let pool_hash = self.get_stake_pool_hash(key)?;
stake_distribution.insert(pool_hash, stakes.snapshot_mark_pool);
```

falcucci replied:
thank you!

falcucci committed:
refactor: stake pool hash fetching and distribution

- simplify `get_stake_pool_hash` function by removing `stakes` argument and adjust the return type.
- now fetch the `stakes.snapshot_mark_pool` directly in the `get_stake_distribution_snapshot` function when inserting into `stake_distribution`.

falcucci committed:
merge 'main'

falcucci committed:
refactor: function and test names

- rename `get_stake_distribution_snapshot` to `get_stake_distribution`
- change the test function name to include `with_fallback` instead of `fallback`

falcucci committed:
add some docs

falcucci committed:
refactor: stake snapshots logic and update imports

- refactor the filter condition in the stake snapshots loop
- update `stakes` import for the changed module structure

falcucci committed:
refactor: streamline query handling in pallas_observer.rs

- refactor imports in `pallas_observer.rs`
- update code references to `localstate::queries_v16` to now use the module specifiers directly
- consolidate initialization and usage of `stakes` struct in tests
- adjust stake values and pools in test data
- rename `localstate::queries_v16::stakesnapshot` to `stakesnapshot`
- refactor matching logic in `match` block for query handling

falcucci committed:
chore: update version to 0.3.5 in cargo.toml

- update the version in cargo.toml from "0.3.4" to "0.3.5"

falcucci committed:
merge 'main'

falcucci committed:
chore: update version metadata to 0.3.7 in cargo.toml

- update the version from `0.3.6` to `0.3.7` in the `mithril-common` package metadata in `cargo.toml`
